sw/device/lib/testing/test_framework/freertos_port.S

```diff
@@ -15,6 +16,40 @@
 
 // -----------------------------------------------------------------------------
 
+/**
+ * Save MEPC to the stack.
+ *
+ * NOTE: this IRQ is synchronous, therefore, we must update the ISR return
+ * address to point to the instruction after the one that triggered this IRQ.
+ * Since we support the RISC-V compressed instructions extension, we need to
+ * check if the two least significant bits of the instruction are
+ * b11 (0x3), which means that the trapped instruction is not compressed,
+ * i.e., the trapped instruction is 32bits = 4bytes. Otherwise, the trapped
+ * instruction is 16bits = 2bytes.
+ */
+
+.balign 4
+.type _process_mepc_on_synchronous_irq, @function
+_process_mepc_on_synchronous_irq:
+  csrr t0, mepc
+  li t1, 0x3
+  and t2, t0, t1
+  beq t2, t1, _32bit_trap_instr
+_16bit_trap_instr:
```

Comment:
I think we've been trying to use `L_` as a prefix for labels. This label is also unused... I think I'd write this much more simply as

```S
  beq t2, t1, 1f
  addi, t0, t0, HALF
  ret
1:
  addi t0, t0, WORD
  ret
```
I'd punt the store to the caller. Doing it in the caller is sketch.

Reply:
Fixed.